### PR TITLE
Facet navigation

### PR DESCRIPTION
**Pull request by banurekha**

### CODE REVIEW

File: spec/curate/functional/func_curate_spec.rb [hunk 1]
```diff
@@ -100,3 +102,65 @@
...
+    end
+    expect(page).not_to have_selector("#ajax-modal")
+  end
+
+  scenario 'Other facets' do
+    visit '/'
+    click_on('Search')
+    within('ul.facets') do
+      facet_listing.each do |facet_name|
+        print "Testing #{facet_name}\n"
```
Comment (jeremyf) [inline spec/curate/functional/func_curate_spec.rb]: Remove `print` statement; Either log this or skip. @h-parekh Thoughts?

File: spec/curate/functional/func_curate_spec.rb [hunk 2]
```diff
@@ -100,3 +102,65 @@
...
+
+  scenario 'Publisher' do
+    visit '/'
+    click_on('Search')
+    expect(page).not_to have_css('#collapse_Publisher.in.collapse')
+    find("a[data-target='#collapse_Publisher']").click
+    expect(page).to have_css('#collapse_Publisher.in.collapse')
+  end
+
+  scenario 'Academic Status' do
```
Comment (jeremyf) [inline spec/curate/functional/func_curate_spec.rb]: This duplicates the "Other facets" scenario.

File: spec/curate/functional/func_curate_spec.rb [hunk 3]
```diff
@@ -100,3 +102,65 @@
...
+        print "Testing #{facet_name}\n"
+        next unless has_content?(facet_name)
+        expect(page).not_to have_css("#collapse_#{facet_name}.in.collapse")
+        find("a[data-target=\"#collapse_#{facet_name}\"]").click
+        expect(page).to have_css("#collapse_#{facet_name}.in.collapse")
+      end
+    end
+  end
+
+  scenario 'Publisher' do
```
Comment (jeremyf) [inline spec/curate/functional/func_curate_spec.rb]: This is duplicated in the "Publisher" scenario.

File: spec/curate/functional/func_curate_spec.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 require 'curate/curate_spec_helper'
 
+facet_listing = ['Type_of_Work', 'Creator', 'Subject', 'Language', 'Publisher', 'Related_Resource(s)', 'Academic_Status'].freeze
```
Comment (jeremyf) [inline spec/curate/functional/func_curate_spec.rb]: Move this variable closer to its use (just above line 137).

Either make this a constant (outside of the scenario) or a variable within the scenario.

File: spec/curate/functional/func_curate_spec.rb [hunk 5]
```diff
@@ -100,3 +102,65 @@
...
+    expect(page).not_to have_selector("#ajax-modal")
+  end
+
+  scenario 'Other facets' do
+    visit '/'
+    click_on('Search')
+    within('ul.facets') do
+      facet_listing.each do |facet_name|
+        print "Testing #{facet_name}\n"
+        next unless has_content?(facet_name)
```
Comment (jeremyf) [inline spec/curate/functional/func_curate_spec.rb]: @h-parekh If we skip a facet, should we log this as a debug item?

**Comment by jeremyf** [comment]
@banurekha There is an outstanding comment and a merge conflict.

File: spec/curate/functional/func_curate_spec.rb [hunk 6]
```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 require 'curate/curate_spec_helper'
 
-feature 'User Browsing:', js: true do
-  scenario 'Load Homepage' do
+feature 'User Browsing', js: true do
+  scenario 'Test start: Load Homepage' do
```
Comment (jeremyf) [inline spec/curate/functional/func_curate_spec.rb]: Please remove the "Test Start:" portion of the spec. This is handled via the logger.

Comment (h-parekh) [inline spec/curate/functional/func_curate_spec.rb]: This has been taken care of in https://github.com/ndlib/QA_tests/commit/7c07da409f2d88f401be2a3f10d53fb0dad96ced

Comment (jeremyf) [inline spec/curate/functional/func_curate_spec.rb]: @h-parekh Except that Banu's PR will overwrite those changes.

**Pull request merged**